packages/angular/projects/ui-angular/src/lib/primitives-poc/button/button-component/button.component.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  HostBinding,
+  Input,
+  OnInit,
+  ViewEncapsulation,
+} from '@angular/core';
+
+@Component({
+  selector: 'button[amplify-button-poc]',
+  templateUrl: './button.component.html',
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AmplifyButtonComponent implements OnInit {
+  @Input() type: 'submit' | 'button' = 'button';
+  @Input() fullWidth: boolean | string = false;
+  @Input() isDisabled: boolean | string = false;
+  @Input() size: 'small' | 'medium' | 'large' = 'medium';
+  @Input() variation: 'primary' | 'default' | 'link' = 'default';
+  @Input() fontWeight: 'normal' | 'bold' | 'lighter' = 'normal';
```

Comment:
Can be a future improvement, but imo it'd be more maintainable and reusable to have this type defined in a shared `types` folder.